python/TestHarness/schedulers/QueueManager.py

```diff
@@ -0,0 +1,403 @@
+        # QueueManager knows best when this is True.
+        if self.__status_check:
+            # Collect all jobs no matter the queue other schedulers are attempting to use.
+            # __status_check is True, so QueueManager knows best ATM.
+            all_jobs = set(status_jobs).union(run_jobs)
+
+            launched_dags = []
+            for job_container in all_jobs:
+                test_unique = self.getUnique(job_container)
+
+                # Only interested in jobs contained in the current session
+                if test_unique not in self.__session_data.keys():
+                    continue
+
+                # Yo dawg, I heard you like DAGs
+                job_dag = job_container.getDAG()
+                original_dag = job_container.getOriginalDAG()
+                reverse_dag = original_dag.reverse_edges()
+
+                # job_dag is unique and shared amongst a group of job_containers. We only want
+                # to work with one of these DAGs per group. So this is how we do it.
+                if job_dag in launched_dags:
+                    continue
+                launched_dags.append(job_dag)
+
+                # Using the original DAG the job_container group began with, and a reverse clone
+                # of that to boot, itereate _backwards_ to try and run postProcess on TestB,
+                # before running postProcess on TestA (because performing postProcess on TestA,
+                # while TestB might still be _running_, can cause bad things for some tests.
+                for original_job_container in reverse_dag.topological_sort():
+                    self.checkStatus(original_job_container)
+
+        # NOTE: The Scheduler attempting to queue a 'status_jobs' while __status_check is False
+        # means this test is one of the skipped variety
+        elif status_jobs:
+            for job_container in status_jobs:
+
+                # We do not want to save silent tests to the session file
+                tester = job_container.getTester()
+                if tester.isSilent():
+                    return
+
+                self.checkStatus(job_container)
+
+        elif run_jobs:
+            for job_container in run_jobs:
+                # store launched jobs so we can use it when TestHarness calls waitFinish.
+                self.__jobs.add(job_container)
+
+                # call derived launchJob methods
+                self.launchJob(job_container)
+
+                # send this job to the status job queue
+                self.queueJobs(status_jobs=[job_container])
+
+                job_dag = job_container.getDAG()
+
+                # Delete this job from the DAG
+                job_dag.delete_node(job_container)
+
+                # Get next job list
+                next_job_group = self.getNextJobGroup(job_dag)
+
+                # run these new jobs
+                self.queueJobs(run_jobs=next_job_group)
+
+    def checkStatus(self, job_container):
```

Comment:
This whole method could go away if you just put the extra dependencies right inside of the "expanded DAG" . In fact, it would just fall back to the statusing method in the Scheduler base class! You think you're obsessed with DAGs?! I'm the one that really wants one DAG to rule them all. Bahahaha